aws/secretsmanager.go

```diff
@@ -0,0 +1,94 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+)
+
+// ReadSecretManagerSecret Reads a secret from the AWS Secrets Manager,
+// given the ARN of the secret and the fieldname (key) to extract from
+// the secret mapping
+func ReadSecretManagerSecret(arn string, fieldname string) ([]byte, error) {
+	secretJSON, err := getSecret(arn)
+	var password string
+	var passwordBytes []byte
+
+	if err != nil {
+		return nil, err
+	}
+
+	if secretJSON != "" {
+		var result map[string]interface{}
+		json.Unmarshal([]byte(secretJSON), &result)
+		password = result[fieldname].(string)
+	}
+
+	passwordBytes = []byte(password)
+	return passwordBytes, err
+}
+
+func getSecret(secretName string) (string, error) {
+	//Create a Secrets Manager client
+	sess, err := session.NewSession()
+	if err != nil {
+		return "", err
+	}
+
+	svc := secretsmanager.New(sess,
+		aws.NewConfig().WithRegion("us-west-2"))
```

Comment:
I _really_ don't like hardcoding the region here. I don't want to fix our PKi infrastructure services once we go multi-region... ;)

Reply:
Can we pass a secretmanager service client in rather than creating it here?

Reply:
(Completely agree here re hardcoding, there are some limitations in the number of interfaces that need to change in order to pass this around.  I'll revisit the change to improve on this.)